### PR TITLE
Refactor /inventory code to expose some helpers

### PR DESCRIPTION
**Pull request by foot**
https://github.com/weaveworks/weave-gitops-enterprise/issues/2776

**What changed?**

- Refactor inventory code
- Expose some of it so we can use it any parts of the code eg: in wge
- Fixes logic to look for children in the parentObjects namespace, _not_ the automation's namespace

**Why was this change made?**

Other similar CRDs that match the flux inventory pattern can share similar code like this

**How was this change implemented?**

**How did you validate the change?**
unit tests

**Release notes**
Expose some inventory directory helper functions 

**Documentation Changes**

### CODE REVIEW

File: core/server/inventory.go [hunk 1]
```diff
@@ -26,6 +28,12 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// an object that can store unstructued and its children
```
[the remark this answers is not in the document]
Comment (bigkevmcd) [reply core/server/inventory.go:31]: But this also suggests that it's maybe an `InventoryTree`?

File: core/server/inventory.go [hunk 2]
```diff
@@ -222,36 +217,34 @@ func getHelmReleaseObjects(ctx context.Context, k8sClient client.Client, helmRel
...
 
-func (cs *coreServer) unstructuredToInventoryEntry(ctx context.Context, clusterName string, k8sClient client.Client, unstructuredObj unstructured.Unstructured, ns string, withChildren bool) (*pb.InventoryEntry, error) {
-	var err error
-
+func unstructuredToInventoryEntry(clusterName string, objWithChildren ObjectWithChildren, clusterUserNamespaces map[string][]v1.Namespace, healthChecker health.HealthChecker) (*pb.InventoryEntry, error) {
+	unstructuredObj := *objWithChildren.Object
 	if unstructuredObj.GetKind() == "Secret" {
-		unstructuredObj, err = sanitizeUnstructuredSecret(unstructuredObj)
+		var err error
+		unstructuredObj, err = SanitizeUnstructuredSecret(unstructuredObj)
```
Comment (bigkevmcd) [inline core/server/inventory.go:224]: `SanitizeUnstructuredSecret` isn't used outside of the `server` package?

File: core/server/inventory.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ranatrk) [reply core/server/inventory.go:31]: @bigkevmcd  wouldn't updating the doc string with specifying its unstructured object be enough? InventoryTree would include an InventoryEntry with its children not an unstructured object so it wouldnt make sense to name it tree.

**Review by foot — COMMENTED** [review]
LGTM!

**Review by ranatrk — APPROVED** [review]

**Pull request merged**